### PR TITLE
COMMERCE-9767 - Adding checks to prevent price tier quantity from bei…

### PR DESCRIPTION
**Pull request by brianikim**
…ng less than 1

https://issues.liferay.com/browse/COMMERCE-9767

@luca-pellizzon 

From @victorcdp ,

The input for the tier price quantity had no restrictions, allowing for the user to input 0 or negative numbers.

Low priority PR

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]
This is fine, but shouldn't we also add a backend validation? If I want to change the value from headless APIs what will happen?

File: modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.price.list.exception;
+
+import com.liferay.portal.kernel.exception.PortalException;
+
+/**
+ * @author Alessio Antonio Rendina
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:20]: I guess this is not really coming from Alessio 😄

**Review by luca-pellizzon — COMMENTED** [review]
much better, just fix the author please

File: modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:20]: haha @luca-pellizzon , sorry I just noticed that

File: modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package com.liferay.commerce.price.list.exception;
+
+import com.liferay.portal.kernel.exception.PortalException;
+
+/**
+ * @author João Victor Cordeiro
+ */
+public class CommerceTierPriceEntryQuantityException
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:22]: I see that this one has been added manually, without going through
modules/apps/commerce/commerce-price-list-service/service.xml
is there a reason for that?

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java (hunk 2 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:22]: hi @luca-pellizzon , I can take care of it. I was not aware we do exception generations through service.xml now.

Comment (brianikim) [reply modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:22]: @luca-pellizzon , the author name can be changed manually yes? I generated the actual file, but it still gives author name for Alessio.

Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:22]: don't need to change the author when it is autogenerated

Comment (brianikim) [reply modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java:22]: So we do commit the exception file with author as Alessio then? when I run buildService the Exception gets generated with him as the author.

Comment (brianikim) [reply modules/apps/commerce/commerce-price-list-api/src/main/java/com/liferay/commerce/price/list/exception/CommerceTierPriceEntryQuantityException.java]: @luca-pellizzon , the changes should be satisfactory now.

**Review by luca-pellizzon — APPROVED** [review]

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
@luca-pellizzon , I rebased as requested. I am re-running ci:test as a result and also because baseline changes were no longer necessary (I want to make sure there's nothing wrong with my local environment).

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:commerce-functional

**Pull request closed without merging**